medicines/doc-index-updater/src/create_manager/mod.rs

```diff
@@ -1,25 +1,65 @@
-use crate::{models::CreateMessage, service_bus_client::create_factory};
+use crate::{
+    models::{CreateMessage, JobStatus},
+    service_bus_client::{create_factory, DocIndexUpdaterQueue},
+    state_manager::StateManager,
+};
+use anyhow::anyhow;
 use azure_sdk_core::errors::AzureError;
 use std::time::Duration;
 use tokio::time::delay_for;
 
+mod sftp_client;
+
 #[tracing::instrument]
-pub async fn create_service_worker() -> Result<String, AzureError> {
-    let mut create_client = create_factory().await?;
+pub async fn create_service_worker(state_manager: StateManager) -> Result<String, anyhow::Error> {
+    tracing::info!("Starting create service worker");
+    let mut create_client = create_factory().await.map_err(|e| {
+        tracing::error!("{:?}", e);
+        anyhow!("Couldn't create the Create Queue")
+    })?;
 
     loop {
-        let message_result: Result<CreateMessage, AzureError> = create_client.receive().await;
-        if let Ok(message) = message_result {
-            tracing::info!("{:?} message receive!", message);
-            retrieve_file_from_sftp(message.document.file_path).await;
-            // TODO: create file in blob storage
-            // TODO: Update index
-            // TODO: Notify state manager
+        match try_process_from_queue(&mut create_client).await {
+            Ok(status) => match status {
+                FileProcessStatus::Success(job_id) => {
+                    let _ = state_manager.set_status(job_id, JobStatus::Done).await?;
+                }
+                FileProcessStatus::NothingToProcess => {}
+            },
+            Err(e) => {
+                tracing::error!("{:?}", e);
+            }
         }
-        delay_for(Duration::from_secs(10)).await;
+        delay_for(Duration::from_secs(5)).await;
+    }
+}
+
+async fn create_file_in_blob(file: Vec<u8>) -> String {
+    format!("create file in blob: {:?}", file)
+}
+
+async fn update_index(blob: String) {
```

Comment:
`create_search_index_entry` or `add_to_search_index`?

Reply:
`add_to_search_index` might be preferable - quite a bit of overloading of `create` already!

Reply:
good call, renamed to `add_to_search_index`